src/main/java/org/opensearch/jdbc/config/HostConnectionProperty.java

```diff
@@ -6,15 +6,23 @@
 
 package org.opensearch.jdbc.config;
 
+import java.net.URI;
+
 public class HostConnectionProperty extends StringConnectionProperty {
     public static final String KEY = "host";
 
     public HostConnectionProperty() {
         super(KEY);
     }
 
+    @Override
+    protected String parseValue(Object value) throws ConnectionPropertyException {
+        String host = super.parseValue(value);
+        // URI class extracts host from the string. It requires a prefix to be set to parse it properly.
+        return URI.create(host.startsWith("http") ? host : "https://" + host).getAuthority();
```

Comment:
`host.startsWith("http") ? host : "https://" + host`
Does host already include "http://"? Because `"https://" + host` would return "https://http://...."

Reply:
But if `host` starts with `http://` it passes the check `host.startsWith("http")` and adds nothing to it